sympy/solvers/decompogen.py

```diff
@@ -1,6 +1,6 @@
 from sympy.core import Function, Pow, sympify
 from sympy.polys import Poly, decompose
-
+from sympy.core.relational import Relational
```

Comment:
I don't think that patching decompogen is the right approach -- I suspect that handling it in periodicity as another case would be better. But I'm having trouble figuring out what the right answer should be: a Relational should give a Boolean result -- T or F -- but the regions in which it does so may be periodic so can we say that sin(x) > 0 is periodic with period of pi? Would it be true that any relational involving `<` or `>` has the same periodicity as `abs(r.lhs - r.rhs)` while those involving `==` or `!=` is just the periodicity of `r.lhs - r.rhs`?

Reply:
I think the period of sin(x)>0 is 2*pi **(sin(x+pi) is -sin(x))**.